File: repos_estudiantes_ALVAREZ_tercera_entrega/GUTIERREZ_GARCIA_CARLA_ALEJANDRA/GENERACIDAD/Ejercicio1/Caja.java
//Crea una clase genérica Caja<T> para guardar algún tipo de objeto 
public class Caja<T> {
    private T contenido;

    // a) Agrega métodos guardar() y obtener()
    public void guardar(T dato) {
        this.contenido = dato;
    }

    public T obtener() {
        return contenido;
    }

    public static void main(String[] args) {
        // b) Crea dos instancias de la caja y almacena 2 datos de diferente tipo
        Caja<String> cTexto = new Caja<>();
        cTexto.guardar("Generacidad practica");

        Caja<Integer> cNumero = new Caja<>();
        cNumero.guardar(45454548);
        // c) Muestra el contenido de las cajas
        System.out.println("Esto muestra texto: " + cTexto.obtener());
        System.out.println("Esto muestra numeros: " + cNumero.obtener());
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/GUTIERREZ_GARCIA_CARLA_ALEJANDRA/GENERACIDAD/Ejercicio1/Caja.py
from typing import Generic, TypeVar

T = TypeVar('T')

class Caja(Generic[T]):
    def __init__(self):
        self.contenido = None

    def guardar(self, objeto: T):
        self.contenido = objeto
        print(f"Objeto guardado: {objeto}")

    def obtener(self) -> T:
        return self.contenido

caja_entero = Caja[int]()
caja_entero.guardar(42)

caja_texto = Caja[str]()
caja_texto.guardar("Hola mundo")

print("caja entero:", caja_entero.obtener())
print("caja texto:", caja_texto.obtener())